crates/emmylua_parser_desc/src/testlib.rs

```diff
@@ -0,0 +1,70 @@
+use crate::util::sort_result;
+use crate::{DescItem, LuaDescParser};
+use emmylua_parser::{
+    LuaAstNode, LuaDocDescription, LuaKind, LuaParser, LuaSyntaxKind, ParserConfig,
+};
+
+pub fn test(code: &str, mut parser: Box<dyn LuaDescParser>, expected: &str) {
+    let tree = LuaParser::parse(code, ParserConfig::default());
+    let Some(desc) = tree
+        .get_red_root()
+        .descendants()
+        .filter(|node| matches!(node.kind(), LuaKind::Syntax(LuaSyntaxKind::DocDescription)))
+        .next()
+    else {
+        panic!("No desc found in {:?}", tree.get_red_root());
+    };
+    let ranges = parser.parse(code, LuaDocDescription::cast(desc).unwrap());
+    let result = format_result(code, ranges);
+    assert_eq!(result.trim(), expected.trim());
```

Comment:
If assert is written inside the function, will the error location always be inside this function when an error occurs?